x-pack/test/api_integration/services/slo_api.ts

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  fetchHistoricalSummaryParamsSchema,
+  FetchHistoricalSummaryResponse,
+} from '@kbn/slo-schema';
+import * as t from 'io-ts';
+import { FtrProviderContext } from '../../functional/ftr_provider_context';
+import { RoleCredentials } from '../../../test_serverless/shared/services';
```

Comment:
Role based testing was introduced in this [PR](https://github.com/elastic/kibana/pull/184654). As part of that PR an `svlUserManager ` service was used within the test itselft to create an api role for admin for serverless.

With current PR where the goal is to have only one test used by both ess and serverless, current approach didn't work. What I tried instead was to move this role-based logic within the `slo_api` service itself The slo api will check the `service` and will create the api key on serverless accordingly.

This worked out well. As you can see below in case of ess it creates all the predefined roles and users, where as on serverless it only creates an api key.

**ess**
<img width="450" alt="Screenshot 2024-07-09 at 14 37 54" src="https://github.com/elastic/kibana/assets/2852703/d6fd2961-d382-4fdd-9812-322dbbbb8559">

<img width="450" alt="Screenshot 2024-07-09 at 14 41 37" src="https://github.com/elastic/kibana/assets/2852703/e67bbb72-f980-47eb-99cf-3a9615a2a636">


**serverless**

<img width="613" alt="Screenshot 2024-07-09 at 14 42 45" src="https://github.com/elastic/kibana/assets/2852703/757eb6ef-7154-4c1c-96e1-8d299d646983">

Let's explore if this approach feels right, and then I will proceed applying the changes to the `alerting_api` service as well. My concern with this approach is that `slo_api` service is added within the `deployment agnostic services`, but internally it uses the config service to check if it is serverless or stateful. What are your thoughts on this? Any other alternatives?